linera-views/src/map_view.rs

```diff
@@ -319,22 +307,32 @@ where
     /// # let context = create_memory_context();
     ///   let mut map = ByteMapView::load(context).await.unwrap();
     ///   map.insert(vec![0,1], String::from("Hello"));
-    ///   let value = map.get_mut(vec![0,1]).await.unwrap().unwrap();
+    ///   let value = map.get_mut(&[0,1]).await.unwrap().unwrap();
     ///   assert_eq!(*value, String::from("Hello"));
     ///   *value = String::from("Hola");
     ///   assert_eq!(map.get(&[0,1]).await.unwrap(), Some(String::from("Hola")));
     /// # })
     /// ```
-    pub async fn get_mut(&mut self, short_key: Vec<u8>) -> Result<Option<&mut V>, ViewError> {
-        self.load_value(&short_key).await?;
-        if let Some(update) = self.updates.get_mut(&short_key) {
-            let value = match update {
+    pub async fn get_mut(&mut self, short_key: &[u8]) -> Result<Option<&mut V>, ViewError> {
+        let update = match self.updates.entry(short_key.to_vec()) {
+            Entry::Vacant(e) => {
+                if self.delete_storage_first || contains_key(&self.deleted_prefixes, short_key) {
+                    None
+                } else {
+                    let key = self.context.base_index(short_key);
+                    let value = self.context.read_value(&key).await?;
+                    value.map(|value| e.insert(Update::Set(value)))
+                }
+            }
+            Entry::Occupied(e) => Some(e.into_mut()),
+        };
+        Ok(match update {
+            None => None,
+            Some(update) => match update {
```

Comment:
nit: `Some(Update::Removed) => None` etc to avoid nested matches (the compiler will re-optimize)